lib/xhp-bootstrap/alert.php

```diff
@@ -33,6 +34,14 @@ protected function render(): XHPRoot {
     $ret->addClass('alert');
     $ret->addClass('alert-'.$this->:use);
 
+    if ($this->:dimiss) {
+        $close =
+          <button type="button" class="close" data-dismiss="alert" aria-label="Close">
+              <span aria-hidden="true">&times;</span>
+          </button>;
+	$this->prependChild($close);
```

Comment:
It looks like you have some spacing issues. The tabbing convention for this project is two spaces; you seem to use both 3 and 4 here.